tests/extend_test.py

```diff
@@ -178,6 +178,11 @@ def fun():
     self.assertTrue(hlo.TokenType.isinstance(op.operands[0].type))
     self.assertTrue(hlo.TokenType.isinstance(op.results[0].type))
 
+  def testEffects(self):
+    def fun():
+      jex.ffi.ffi_call("test_ffi", (), has_side_effect=True)
+    self.assertIn("test_ffi", jax.jit(fun).lower().as_text())
```

Comment:
Probably you should verify that it doesn't get DCEd from the compiled IR, not the input to the compiler?

Or at least check the input IR has the side effect annotation?

Reply:
Good suggestion! Done.